### DISCUSSION TITLE
Exporting call peaks result as bed graph file

### DISCUSSION THREAD

**Discussion by yasamanrezvani** [stuart-lab/signac#559] [Q&A] [answered]
Hi, 

Thanks for this nice package. I am trying too use callpeaks() function to get the correct number of peaks. The function output is a Grange object that I can turn it into a data frame. What I would also like to have is the bed graph format. I realized that I can use the additional.args to pass --bdg to MACS2. but I don't know how to do this in R. Can you guide me on this. thanks.

 ` peaks.MCS <- CallPeaks(
  object =  "fragments.tsv.gz"),
  macs2.path = "/opt/anaconda3/bin/macs2", 
  extsize = 100)`

**Comment by timoast** [comment, marked as answer]
The bedgraph format is similar to the format of the `GRanges` object that's returned by the function, except that you'd need to construct the bedgraph header. If you're not using the cell-type-specific peak calling functionality in Signac, I'd recommend instead running MACS2 outside of R to get the bedgraph output.

**Comment by tilofrei** [comment]
You could use the option `cleanup`: `CallPeaks(.., cleanup=FALSE)`. That way you find the resulting peak files (bed/narrowPeak/xls) in `tempdir()` and can copy them elsewhere.

Another way is exporting them from R (that way you have access to the merged peaks):
```
peaks <- CallPeaks(..)
rtracklayer::export.bed(peaks, "./path/my_peaks.bed")
```
